Add verification instructions for CGP61 (#165)
celo-org/governance: CGPs/cgp-0061.md

If for whatever reason pausing the pool is problematic, two CGPs can be issued. The first is just to remove liquidity, and the second to burn/transfer with the known balances that result from the liquidity removal. 
 
# Verification

## Before exection 

Before approving/voting: Fetch the on-chain proposal and verify that the changes proposed above match the changes proposed on-chain:
```

> celocli governance:show --proposalID <TBD> --node https://forno.celo.org

Reviewer: ```suggestion
celocli governance:show --proposalID 68 --node https://forno.celo.org
```

```

### Transactions 

#### 1. StableTokenProxy._setImplemention

The new StableToken implementation is deployed at `0x41a2887d4C4D96C9E1a3505CF4553Fd0b1380F13`.  
The code is verified [on CeloScan](todo:addlink), and the diff between this implementation and the current one, can be seen [here](todo:addlink).

#### 2. StableToken.mint

The two params represent who to mint to and how much:

> - Who should be governance, the address can be verified

Reviewer: ```suggestion
- Who should be governance, the address can be verified by looking it up on explorer.celo.org or celoscan.io. 
```

- The amount should be 5M cUSD, with 18 decimals, so 5 * 1e6 * 1e18 = 5 * 1e23

#### 3. StableProxyToken._setImplementation

After the mint, we revert the implementation to the existing one (before the proposal).
This address can be verified by calling `_getImplementation` on [StableTokenProxy](todo:addlink).

#### 4. StableToken.approve

In order to deposit liquidity in the pool we first need to allow the pool to transfer tokens from governance.

> This parameters of this call are the pool address and the amount to set approval for:

Reviewer: ```suggestion
The parameters of this call are the pool address and the amount to set approval for:
```

> - The pool address

Reviewer: ```suggestion
- The pool address (todo: add pool address)
```

- 5M cUSD, with 18 decimals, so 5 * 1e23 

#### 5. USDC.approve 

Same as (4) but for setting USDC approval. The [USDC contract](todo:addlink) can be verified on-chain. 
The parameters should be:

> - The pool address

Reviewer: ```suggestion
- The pool address (todo: add pool address)
```

- 5M USDC, with 6 decimals, so 5 * 1e6 * 1e6 = 5 * 1e12

> The number of decimals for USDC can be verified [here](todo:addlink)

#### 6. Pool.addLiquidity

> Adding liquidity to the pool requires thre parameters:

Reviewer: ```suggestion
Adding liquidity to the pool requires three parameters:
```

> - The amounts to add as an array, the order of which depends on the indexes in the pool, in this case first the cUSD amount and then the USDC amount, number of decimals follows what is discussed in steps (4) and (5)

Reviewer: ```suggestion
- The amounts to add as an array. The order of this array depends on the indices in the pool, in this case the cUSD amount first and then the USDC amount. Number of decimals follows what is discussed in steps (4) and (5).
```

> - The slippage parameters, minimum amount of LP tokens to be minted as a result of the deposit, we can safely set this to 0 in the case of a constant sum market maker.

Reviewer: ```suggestion
- The slippage parameters, that is the minimum amount of LP tokens to be minted as a result of the deposit. We can safely set this to 0 in the case of a constant sum market maker.
```
